### ISSUE TITLE
docker:// transport requires "docker login" and doesn't obey --username and --password

### ISSUE THREAD

**Issue by cyphar** [projectatomic/skopeo#253]
If you try to do a `skopeo copy` to a Docker registry repository that doesn't exist you get an error:

```
% skopeo copy docker://opensuse docker://cyphar/nonexistant
Getting image source signatures
Copying blob sha256:467db25190688bc9dc1d5fd6dbced1ac56f55d93a942987f448e62ad2614e46e
 0 B / 46.97 MB [--------------------------------------------------------------]
FATA[0004] Error writing blob: not authorized to read from destination repository cyphar/nonexistent
```

The above example does work if you first do a `docker push cyphar/nonexistent` but that's quite annoying. I'm not sure if this should be transparently handled by containers/image or if we should be implementing this policy.

**Comment by runcom** [comment]
@cyphar I'm not sure which version you are using but that works perfectly fine here:
```
13:26:44 [github.com/projectatomic/skopeo] ‹master› ./skopeo copy docker://busybox docker://runcom/doesntexistfdasfkjlfalfas
Getting image source signatures
Copying blob sha256:56bec22e355981d8ba0878c6c2f23b21f422f30ab0aba188b54f1ffeff59c190
 652.49 KB / 652.49 KB [=======================================================]
Copying config sha256:e02e811dd08fd49e7f6032625495118e63f597eb150403d02e3238af1df240ba
 1.43 KB / 1.43 KB [===========================================================]
Writing manifest to image destination
Storing signatures
```
Have you logged in?

**Comment by cyphar** [comment]
Yeah, I'm using `--username` and `--password`. This happened on `master`.

**Comment by runcom** [comment]
> Yeah, I'm using --username and --password. This happened on master.

you don't have --username and --password in you reproducer...

**Comment by runcom** [comment]
even with both --username and --password:
```
13:29:31 [github.com/projectatomic/skopeo] ‹master› ./skopeo --username runcom --password MYAWESOMEPASS copy docker://busybox docker://runcom/doesntexistfdasfkjlfalfas
Getting image source signatures
Copying blob sha256:56bec22e355981d8ba0878c6c2f23b21f422f30ab0aba188b54f1ffeff59c190
 0 B / 652.49 KB [-------------------------------------------------------------]
Copying config sha256:e02e811dd08fd49e7f6032625495118e63f597eb150403d02e3238af1df240ba
 0 B / 1.43 KB [---------------------------------------------------------------]
Writing manifest to image destination
Storing signatures
```

**Comment by cyphar** [comment]
Sorry, I removed it from the reproducer. Here's the _actual_ thing I'm using;

```
% ../skopeo/skopeo --username cyphar --password PASSWORD copy oci:opensuse:new docker://cyphar/opensuse_playground:lol
Getting image source signatures
Copying blob sha256:467db25190688bc9dc1d5fd6dbced1ac56f55d93a942987f448e62ad2614e46e
 0 B / 46.97 MB [--------------------------------------------------------------]
FATA[0004] Error writing blob: not authorized to read from destination repository cyphar/nonexistent
```

I'm patching `vendor/` by copying the current master of containers/image because I needed to test something for a blog post.

**Comment by runcom** [comment]
so, username/password flags aren't enabled in containers/image yet https://github.com/containers/image/blob/master/docker/docker_client.go#L288. This is a different issue though. This isn't requiring to have the repo already created on the registry. This is just a matter of authentication and authorization. Could you rephrase the title of this issue?

**Comment by runcom** [comment]
I'll try to patch containers/image now.

**Comment by cyphar** [comment]
Ah, okay. It looks like the issue was related to the fact that I hadn't used `docker login` on this machine before. Because after I logged in to push the original image, my reproducer now works. Thanks runcom.

By the way in Australia I keep getting `i/o timeout` errors. :cry:
